unitTests/Repartitioner.C

```diff
@@ -58,64 +63,314 @@ public:
+        detail::exchange_span_ranks(exec, ranks_per_gpu, spans, ranks);
+
+    std::vector<label> res_spans_begin{};
+    std::vector<label> res_spans_end{};
+
+    for (auto span : new_spans) {
+        res_spans_begin.push_back(span.begin);
+        res_spans_end.push_back(span.end);
+    }
+
+    // Assert
+    ASSERT_EQ(new_ranks, exp_ranks[ranks_per_gpu][rank]);
+    ASSERT_EQ(res_spans_begin, exp_spans_begin[ranks_per_gpu][rank]);
+    ASSERT_EQ(res_spans_end, exp_spans_end[ranks_per_gpu][rank]);
 }
 
+TEST_P(RepartitionerFixture1D, can_repartition_sparsity_pattern_1D_for_n_ranks)
+{
+    // Arrange
+    label ranks_per_gpu = GetParam();
+    auto repartitioner = Repartitioner(local_size, ranks_per_gpu, 0, exec);
+    auto ref_exec = exec.get_ref_exec();
+
+    std::vector<label> ranks{rank};
+    auto local_sparsity = std::make_shared<SparsityPattern>(
+        ref_exec, gko::dim<2>{2, 2}, rows, cols, mapping, spans, ranks);
+
+    auto non_local_sparsity = std::make_shared<SparsityPattern>(
+        ref_exec, gko::dim<2>{2, non_local_ranks[rank].size()},
+        non_local_rows[rank], non_local_cols[rank], non_local_mapping[rank],
+        non_local_spans[rank], non_local_ranks[rank]);
+
+    std::map<label, std::vector<label>> exp_local_nnz;
+    exp_local_nnz.emplace(1, std::vector<label>{4, 4, 4, 4});
+    exp_local_nnz.emplace(2, std::vector<label>{10, 0, 10, 0});
+    exp_local_nnz.emplace(4, std::vector<label>{22, 0, 0, 0});
+
+    std::map<label, std::vector<label>> exp_non_local_nnz;
+    exp_non_local_nnz.emplace(1, std::vector<label>{1, 2, 2, 1});
+    exp_non_local_nnz.emplace(2, std::vector<label>{1, 0, 1, 0});
+    exp_non_local_nnz.emplace(4, std::vector<label>{0, 0, 0, 0});
+
+    std::map<label, vec_vec> exp_local_rows;
+    exp_local_rows.emplace(1, vec_vec{rows, rows, rows, rows});
+    exp_local_rows.emplace(2, vec_vec{{0, 0, 1, 1, 2, 2, 3, 3, 1, 2},
+                                      {},
+                                      {0, 0, 1, 1, 2, 2, 3, 3, 1, 2},
+                                      {}});
+    exp_local_rows.emplace(4, vec_vec{{0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5,
+                                       5, 6, 6, 7, 7, 1, 2, 3, 4, 5, 6},
+                                      {},
+                                      {},
+                                      {}});
+
+    std::map<label, vec_vec> exp_local_cols;
+    exp_local_cols.emplace(1, vec_vec{cols, cols, cols, cols});
+    exp_local_cols.emplace(2, vec_vec{{0, 0, 1, 1, 2, 2, 3, 3, 1, 2},
+                                      {},
+                                      {0, 0, 1, 1, 2, 2, 3, 3, 1, 2},
+                                      {}});
+    exp_local_cols.emplace(4, vec_vec{{0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5,
+                                       5, 6, 6, 7, 7, 1, 2, 3, 4, 5, 6},
+                                      {},
+                                      {},
+                                      {}});
```

Comment:
In the case of `ranks per gpu` =2 or 4, ` exp_local_cols` and `exp_local_rows` should be different.

Reply:
fixed in [#impl/integration_06](https://github.com/hpsim/OGL/tree/impl/integration_06)